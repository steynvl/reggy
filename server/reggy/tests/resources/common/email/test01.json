{
  "type": "Email address",
  "information": {
    "username": "Allow any user name",
    "domainName": "Allow any domain name",
    "mailtoPrefix": "No prefix",
    "specificUserNamesOnly": "",
    "domainOnSpecificTld": "",
    "anySubDomainOnSpecificDomain": "",
    "specificDomainsOnly": ""
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Start of word",
    "endRegexMatchAt": "End of word",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}