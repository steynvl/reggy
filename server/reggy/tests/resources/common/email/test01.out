Java
Pattern regex = Pattern.compile("\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+\\b");
\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+\\b
Perl
my $regex = /\b[\d!#$%&'*+\/=?_`a-z{|}~^-]++(?:\.[\d!#$%&'*+\/=?_`a-z{|}~^-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+\b/;
\b[\d!#$%&'*+\/=?_`a-z{|}~^-]++(?:\.[\d!#$%&'*+\/=?_`a-z{|}~^-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+\b
POSIX
\b[\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\.[\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+\b
\b[\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\.[\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+\b
Python
regex = re.compile(r'\b[\d!#$%&\'*+/=?_`a-z{|}~^-]+(?:\.[\d!#$%&\'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b')
\b[\d!#$%&\'*+/=?_`a-z{|}~^-]+(?:\.[\d!#$%&\'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b
JavaScript
const regex = /\b[\d!#$%&'*+\/=?_`a-z{|}~^-]+(?:\.[\d!#$%&'*+\/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b/;
\b[\d!#$%&'*+\/=?_`a-z{|}~^-]+(?:\.[\d!#$%&'*+\/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b
PHP
$regex = '/\b[\d!#$%&'*+\/=?_`a-z{|}~^-]++(?:\.[\d!#$%&'*+\/=?_`a-z{|}~^-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+\b/';
\b[\d!#$%&'*+\/=?_`a-z{|}~^-]++(?:\.[\d!#$%&'*+\/=?_`a-z{|}~^-]+)*@(?:[\da-z-]++\.)+[a-z]{2,63}+\b
Golang
regex, _ := regexp.Compile("\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]+(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]+\\.)+[a-z]{2,63}\\b")
\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]+(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]+\\.)+[a-z]{2,63}\\b
Rust
let re = Regex::new(r"\b[\d!#$%&'*+/=?_`a-z{|}~^-]+(?:\.[\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b").unwrap();
\b[\d!#$%&'*+/=?_`a-z{|}~^-]+(?:\.[\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b
C#
var re = new Regex(@"\b[\d!#$%&'*+/=?_`a-z{|}~^-]+(?:\.[\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b");
\b[\d!#$%&'*+/=?_`a-z{|}~^-]+(?:\.[\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\da-z-]+\.)+[a-z]{2,63}\b
Scala
val re = "\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+\\b".r
\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+\\b
Kotlin
val regex = Regex("\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+\\b")
\\b[\\d!#$%&'*+/=?_`a-z{|}~^-]++(?:\\.[\\d!#$%&'*+/=?_`a-z{|}~^-]+)*@(?:[\\da-z-]++\\.)+[a-z]{2,63}+\\b